eng/pipelines/templates/steps/vcpkg.yml

```diff
@@ -1,13 +1,15 @@
 steps:
   - pwsh: |
       Write-Host "##vso[task.setvariable variable=VCPKG_BINARY_SOURCES_SECRET;issecret=true;]clear;x-azblob,https://cppvcpkgcache.blob.core.windows.net/public-vcpkg-container,,read"
+      Write-Host "##vso[task.setvariable variable=X_VCPKG_ASSET_SOURCES_SECRET;issecret=true;]clear;x-azurl,https://cppvcpkgcache.blob.core.windows.net/public-vcpkg-asset-container/,,read"
     displayName: Set Vcpkg Variables
 
-  - task: PowerShell@2
-    inputs:
-      pwsh: true
-      targetType: filePath
-      filePath: eng/scripts/Set-VcpkgWriteModeCache.ps1
-      arguments: -StorageAccountKey '$(cpp-vcpkg-cache-storage-key)'
+  - task: AzurePowerShell@5
     displayName: Set Vcpkg Write-mode Cache
     condition: and(succeeded(), eq(variables['System.TeamProject'], 'internal'))
```

Comment:
See https://github.com/Azure/azure-sdk-for-cpp/pull/5603/files you will need to make this a template condition otherwise it will not work for public builds.